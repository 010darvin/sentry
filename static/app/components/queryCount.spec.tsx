import {render} from 'sentry-test/reactTestingLibrary';

import QueryCount from 'sentry/components/queryCount';

describe('QueryCount', function () {
  it('displays count when no max', function () {
    const {container} = render(<QueryCount count={5} />);
    expect(container).toMatchSnapshot();
  });
  it('displays count when count < max', function () {
    const {container} = render(<QueryCount count={5} max={500} />);
    expect(container).toMatchSnapshot();
  });

  it('does not render if count is 0', function () {
    const {container} = render(<QueryCount count={0} />);
    expect(container).toBeEmptyDOMElement();
  });

  it('can render when count is 0 when `hideIfEmpty` is false', function () {
    const {container} = render(<QueryCount count={0} hideIfEmpty={false} />);
    expect(container).toMatchSnapshot();
  });

  it('displays max count if count >= max', function () {
    const {container} = render(<QueryCount count={500} max={500} />);
    expect(container).toMatchSnapshot();
  });
});
