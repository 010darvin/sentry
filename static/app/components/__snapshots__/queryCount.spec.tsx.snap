// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryCount can render when count is 0 when \`hideIfEmpty\` is false 1`] = `
<div>
  <span>
    <span>
      (
    </span>
    <span>
      0
    </span>
    <span>
      )
    </span>
  </span>
</div>
`;

exports[`QueryCount displays count when count < max 1`] = `
<div>
  <span>
    <span>
      (
    </span>
    <span>
      5
    </span>
    <span>
      )
    </span>
  </span>
</div>
`;

exports[`QueryCount displays count when no max 1`] = `
<div>
  <span>
    <span>
      (
    </span>
    <span>
      5
    </span>
    <span>
      )
    </span>
  </span>
</div>
`;

exports[`QueryCount displays max count if count >= max 1`] = `
<div>
  <span>
    <span>
      (
    </span>
    <span>
      500+
    </span>
    <span>
      )
    </span>
  </span>
</div>
`;
